plugins/ingest/src/main/java/org/elasticsearch/plugin/ingest/IngestBootstrapper.java

```diff
@@ -126,6 +141,39 @@ boolean validClusterState(ClusterState state) {
         }
     }
 
+    boolean isIngestTemplateInstallationRequired(MetaData metaData) {
+        if (metaData.getTemplates().containsKey(INGEST_INDEX_TEMPLATE_NAME)) {
+            logger.trace("not installing ingest index template, because it already is installed");
+            return false;
+        }
+        return true;
+    }
+
+    void forkAndInstallIngestIndexTemplate() {
+        threadPool.executor(ThreadPool.Names.GENERIC).execute(() -> {
+            try {
+                installIngestIndexTemplate();
+            } catch (IOException e) {
+                logger.error("Failed to install .ingest index template", e);
```

Comment:
This method will only be invoked if ingest is running and the ingest template gets removed. (for example, a user deletes all his index templates). In this case we try to put the index template back if that fails we try again on the next cluster state update. (on each modification to the cluster the `clusterChanged` gets invoked)

In the case that we're not started and adding the index template fails we don't start the `PipelineStore` and on the next cluster state update we try to add the template and start the `PipelineStore` again.